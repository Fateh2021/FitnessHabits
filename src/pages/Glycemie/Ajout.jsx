import firebaseDb from "../../firebaseConfig"
import React, { useState } from 'react';
import { arrowBack } from "ionicons/icons";
import {
    IonContent,
    IonHeader,
    IonPage,
    IonTitle,
    IonToolbar,
    IonFab,
    IonFabButton,
    IonIcon,
    IonCard,
    IonButton,
    IonCardContent,
    IonDatetime,
    IonCardHeader,
    IonCardTitle,
    IonSelectOption,
    IonSelect
} from "@ionic/react";
import '../Tab1.css'


const Ajout = () => {

    const initialValues = {
        id: '',
        jour: [],
        heure: '',
        notifierAvant: ''    }

    var [values, setValues] = useState(initialValues)

    const handleInputChange = e => {
        var { name, value } = e.target
        setValues({
            ...values,
            [name]: value
        })
    }

    const handleFormSubmit = e => {
        e.preventDefault();
        add(values)
    }

    const add = obj => {
        firebaseDb.child('NotificationGlycemie').push(
            obj,
            err => {
                if (err)
                    console.log(err);
                else {
                    window.location.href = "/glycemie"
                }
            }
        )
    }

    const jour = ["Lundi", "Mardi", "Mercredi", "Jeudi", "Vendredi", "Samedi", "Dimanche"];

    return (
        <div>
            <IonPage>

                <IonHeader className="notGlyAdd">
                    <IonToolbar>
                        <IonTitle class="titre">Nouveau horaire</IonTitle>
                    </IonToolbar>

                    <IonFab class="arrow" vertical="top" horizontal="end" slot="fixed">
                        <IonFabButton routerDirection="back" href="/glycemie" size="small">
                            <IonIcon icon={arrowBack} />
                        </IonFabButton>
                    </IonFab>
                </IonHeader>

                <IonContent>

                    <form onSubmit={handleFormSubmit}>

                        <IonCard>
                            <IonCardHeader>
                                <IonCardTitle>Jour</IonCardTitle>
                            </IonCardHeader>
                            <IonCardContent>
                                <IonSelect multiple={true} name="jour" onIonChange={handleInputChange} value={values.jour} >
                                    {jour.map((option, i) => (
                                    <IonSelectOption value={option} key={i}>{option}</IonSelectOption>))} 
                                </IonSelect>
                            </IonCardContent>

                        </IonCard>

                        <IonCard>
                            <IonCardHeader>
                                <IonCardTitle>Heure</IonCardTitle>
                            </IonCardHeader>
                            <IonCardContent>
                                <IonDatetime display-format="h:mm A" name="heure" picker-format="h:mm: A" onIonChange={handleInputChange} value={values.heure}></IonDatetime>
                            </IonCardContent>
                        </IonCard>

                        <IonCard>
                            <IonCardHeader>
                                <IonCardTitle>Notifier avant:</IonCardTitle>
                            </IonCardHeader>
                            <IonCardContent>
                                <IonDatetime name="notifierAvant" display-format="mm" picker-format="mm" value={values.notifierAvant} onIonChange={handleInputChange}></IonDatetime>
                            </IonCardContent>
                        </IonCard>

                        <IonButton expand="block" type="submit">Ajouter</IonButton>
                    </form>
                </IonContent>
            </IonPage>
        </div>
    );


};

export default Ajout;
