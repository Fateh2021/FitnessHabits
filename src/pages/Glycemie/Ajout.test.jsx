 import React from 'react';
 import renderer from 'react-test-renderer';
 import Ajout from './Ajout.jsx'
 
 
 test('Ajout', ()=> {
 const component = renderer.create(<Ajout/>);
 const tree = component.toJSON();
 expect(tree).toMatchSnapshot();
 
 });
 
 
