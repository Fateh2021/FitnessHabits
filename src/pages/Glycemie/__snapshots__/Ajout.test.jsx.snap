// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ajout 1`] = `
<div>
  <div
    className="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-title>
          Nouveau horaire
        </ion-title>
      </ion-toolbar>
      <ion-fab>
        <ion-fab-button
          onClick={[Function]}
        >
          <ion-icon />
        </ion-fab-button>
      </ion-fab>
    </ion-header>
    <ion-content>
      <form
        onSubmit={[Function]}
      >
        <ion-card
          onClick={[Function]}
        >
          <ion-card-header>
            <ion-card-title>
              Jour
            </ion-card-title>
          </ion-card-header>
          <ion-card-content>
            <ion-select>
              <ion-select-option>
                Lundi
              </ion-select-option>
              <ion-select-option>
                Mardi
              </ion-select-option>
              <ion-select-option>
                Mercredi
              </ion-select-option>
              <ion-select-option>
                Jeudi
              </ion-select-option>
              <ion-select-option>
                Vendredi
              </ion-select-option>
              <ion-select-option>
                Samedi
              </ion-select-option>
              <ion-select-option>
                Dimanche
              </ion-select-option>
            </ion-select>
          </ion-card-content>
        </ion-card>
        <ion-card
          onClick={[Function]}
        >
          <ion-card-header>
            <ion-card-title>
              Heure
            </ion-card-title>
          </ion-card-header>
          <ion-card-content>
            <ion-input />
          </ion-card-content>
        </ion-card>
        <ion-card
          onClick={[Function]}
        >
          <ion-card-header>
            <ion-card-title>
              Notifier avant:
            </ion-card-title>
          </ion-card-header>
          <ion-card-content>
            <ion-input />
          </ion-card-content>
        </ion-card>
        <ion-button
          onClick={[Function]}
        >
          Ajouter
        </ion-button>
      </form>
    </ion-content>
  </div>
</div>
`;
